<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Toy examples of fitting mJAM</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="reference.html">Reference</a>
</li>
<li>
  <a href="examples.html">Software and Tutorial</a>
</li>
<li>
  <a href="contact.html">Contact</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Toy examples of fitting mJAM</h1>

</div>


<div id="installation" class="section level2">
<h2>Installation</h2>
<p>Please install the package <code>hJAM</code> (Hierarchical Joint
Analysis of Marginal Summary Statistics) from github as
<code>mJAM</code> is developed as part of the <code>hJAM</code> package.
The <code>hJAM</code> package also contains other
<code>JAM</code>-family approaches: <code>hJAM</code> (for MR and TWAS)
and <code>SHAJAM</code> (for MR and TWAS with high-dimensional
intermediates).</p>
<pre class="r"><code>devtools::install_github(&quot;USCbiostats/hJAM&quot;)</code></pre>
</div>
<div id="special-notes-for-macos-users" class="section level2">
<h2>Special notes for macOS users</h2>
<p>For macOS users with apple silicon (M1 Macs), please install R
version 4.2-arm64 from <a href="https://mac.r-project.org/"
class="uri">https://mac.r-project.org/</a> . This is a specific build
for compatibility with the M1 arm-based Macs. One of crucial
dependencies in <code>mJAM</code>, <code>Rmpfr</code>, is not yet
compatible with Intel-based R versions running on M1 Macs and it will
cause R/Rstudio to abort. But using version 4.2-arm64 will solve this
issue.</p>
<pre class="r"><code>library(hJAM)
library(dplyr) ## for the use of piping operator %&gt;% </code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
</div>
<div id="example-1-complete-snp-information"
class="section level2 tabset">
<h2 class="tabset">Example 1: Complete SNP information</h2>
<p>In the first example, we’ll illustrate the usage of
<code>mJAM-SuSiE</code> and <code>mJAM-Forward</code> with a simulated
dataset, consisting of 3 ancestry groups and 50 SNPs. These 50 SNPs
consist of 5 LD blocks, each with 10 SNPs and uniform pairwise LD
between them. The first SNP, named “rs1”, is the only true causal
SNP.</p>
<p>First, let’s load the marginal summary statistics of these 50
SNPs.</p>
<pre class="r"><code># Load marginal betas data 
MargBeta &lt;- read.table(&quot;./Example-Data/Marg_1.txt&quot;)
names(MargBeta) &lt;- c(&quot;SNP&quot;, &quot;MargBeta_P1&quot;, &quot;MargSEBeta_P1&quot;, 
                      &quot;MargBeta_P2&quot;, &quot;MargSEBeta_P2&quot;,
                      &quot;MargBeta_P3&quot;, &quot;MargSEBeta_P3&quot;)
head(MargBeta)</code></pre>
<pre><code>##   SNP MargBeta_P1 MargSEBeta_P1 MargBeta_P2 MargSEBeta_P2 MargBeta_P3
## 1 rs1  0.06054038    0.02489064 0.077964524    0.02068987 0.026311626
## 2 rs2 -0.01068437    0.02126704 0.040433268    0.01813179 0.024127159
## 3 rs3  0.05264446    0.02153163 0.037045359    0.01838805 0.019806117
## 4 rs4 -0.01167104    0.02121476 0.009254768    0.01826285 0.006243936
## 5 rs5  0.03411272    0.02145365 0.032849181    0.01819438 0.024697970
## 6 rs6  0.02948659    0.02152562 0.049799131    0.01836573 0.016761096
##   MargSEBeta_P3
## 1    0.02020833
## 2    0.01781806
## 3    0.01786128
## 4    0.01792291
## 5    0.01774054
## 6    0.01782569</code></pre>
<p>Then load the effect allele frequencies (not necessarily the minor
alleles).</p>
<pre class="r"><code># Load effective allele frequency data
EAF &lt;- read.table(&quot;./Example-Data/EAF_1.txt&quot;)
names(EAF) &lt;- c(&quot;SNP&quot;, &quot;EAF_P1&quot;, &quot;EAF_P2&quot;, &quot;EAF_P3&quot;)
head(EAF)</code></pre>
<pre><code>##   SNP EAF_P1 EAF_P2 EAF_P3
## 1 rs1 0.2010 0.4072 0.6075
## 2 rs2 0.2374 0.4205 0.5825
## 3 rs3 0.2329 0.4102 0.5935
## 4 rs4 0.2394 0.4151 0.5850
## 5 rs5 0.2386 0.4158 0.5851
## 6 rs6 0.2344 0.4191 0.5895</code></pre>
<p>Finally one reference dosage matrix for each ancestry group.</p>
<pre class="r"><code># Load individual-level SNP data from reference panel
RefDosage_P1 &lt;- read.table(&quot;./Example-Data/Dosage_S1_1.txt&quot;, header = TRUE)
RefDosage_P2 &lt;- read.table(&quot;./Example-Data/Dosage_S2_1.txt&quot;, header = TRUE)
RefDosage_P3 &lt;- read.table(&quot;./Example-Data/Dosage_S3_1.txt&quot;, header = TRUE)
RefDosage_P1[1:5, 1:5]</code></pre>
<pre><code>##   rs1 rs2 rs3 rs4 rs5
## 1   0   0   0   0   0
## 2   0   0   0   0   0
## 3   0   1   0   0   1
## 4   1   1   1   0   1
## 5   1   1   1   0   1</code></pre>
<pre class="r"><code>RefDosage_P2[1:5, 1:5]</code></pre>
<pre><code>##   rs1 rs2 rs3 rs4 rs5
## 1   1   2   1   2   2
## 2   0   0   0   0   0
## 3   0   0   0   0   2
## 4   0   0   0   0   0
## 5   0   0   0   0   0</code></pre>
<pre class="r"><code>RefDosage_P3[1:5, 1:5]</code></pre>
<pre><code>##   rs1 rs2 rs3 rs4 rs5
## 1   0   1   0   1   0
## 2   1   0   0   1   1
## 3   1   1   1   1   0
## 4   2   2   1   1   2
## 5   1   1   2   0   1</code></pre>
<p>Now we’ll use the marginal summary statistics (<code>beta</code>,
<code>se</code>, <code>EAF</code> and <code>reference dosage</code>) to
run <code>mJAM-SuSiE</code> and <code>mJAM-SuSiE</code>.</p>
<div id="mjam-forward" class="section level3">
<h3>mJAM-Forward</h3>
<pre class="r"><code># --- Fit mJAM_Forward 
Ex1_Forward_fit &lt;- mJAM_Forward(N_GWAS = c(5000, 5000, 5000), 
                                X_ref = list(RefDosage_P1,RefDosage_P2,RefDosage_P3), 
                                Marg_Result = MargBeta,
                                EAF_Result = EAF, 
                                condp_cut = 0.05/50, # Bonferroni-corrected p-val
                                within_pop_threshold = 0.50, 
                                across_pop_threshold = 0.20,
                                coverage = 0.95)</code></pre>
<pre><code>## No. 1 selected index SNP: rs1</code></pre>
<pre><code>## 10 SNPs got pruned. 41 SNPs left.</code></pre>
<pre><code>## Continue to next round of index SNP selection.</code></pre>
<pre><code>## Analysis ended.</code></pre>
<p><code>mJAM_Forward</code> outputs a list with three tables:
<code>index</code>, <code>cs</code> and <code>marginal_est</code>.</p>
<pre class="r"><code># Here&#39;re the selected index SNP(s)
Ex1_Forward_fit$index</code></pre>
<pre><code>## # A tibble: 1 × 6
##   SNP   b_joint b_joint_var cond_log10p final_log10p  pcut
##   &lt;chr&gt;   &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
## 1 rs1    0.0544    0.000156       -4.86        -4.86 0.001</code></pre>
<p><code>index</code> table lists all the index SNP(s) being selected
(<code>SNP</code>), along with their log10(p-value) conditional on all
SNP(s) above (<code>cond_log10p</code>), the log10(p-value) conditional
on all other index SNP(s) (<code>final_log10p</code>), and the p-value
threshold used in this analysis (<code>pcut</code>).
<code>b_joint</code> is the conditional effect estimates in the final
model, and <code>b_joint_var</code> is the squared standard error of
<code>b_joint</code>.</p>
<p>We choose to report log10(p-value) instead of p-value on the original
scale because R does not support p-value precision smaller than around
1e-300. To avoid ambiguous results under large-effect regions, we
calculate p-values on the log/log10 scales and report these.</p>
<pre class="r"><code># Here&#39;re the credible sets SNPs
Ex1_Forward_fit$cs %&gt;% filter(CS_in == T)</code></pre>
<pre><code>## # A tibble: 3 × 7
##   index_SNP CS_SNP Post_Model_Prob Post_Med_Prob SD_Post_CS_Prob CumSum_…¹ CS_in
##   &lt;chr&gt;     &lt;chr&gt;            &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;     &lt;dbl&gt; &lt;lgl&gt;
## 1 rs1       rs1             1              1              0.768      0.768 TRUE 
## 2 rs1       rs8             0.207          0.826          0.131      0.899 TRUE 
## 3 rs1       rs10            0.0859         0.850          0.0560     0.955 TRUE 
## # … with abbreviated variable name ¹​CumSum_Prob</code></pre>
<p><code>cs</code> table records the posterior probabilities of all SNPs
being considered for credible set SNPs. <code>Post_Model_Prob</code>
stands for posterior model probability which quantifies the marginal
association between <code>CS_SNP</code> and its <code>index_SNP</code>.
<code>Post_Med_Prob</code> stands for posterior mediation probability
which quantifies the mediation effect of the <code>CS_SNP</code> through
<code>index_SNP</code>. Stronger LD between <code>CS_SNP</code> and
<code>index_SNP</code> often leads to larger <code>Post_Med_Prob</code>.
By taking the product of <code>Post_Model_Prob</code> and
<code>Post_Med_Prob</code> and then standardizing, we have
<code>SD_Post_CS_Prob</code> which we use to determine the credible
set.</p>
<p>Here we only print out those that are selected in the 95% mJAM
credible set (rs1, rs8 and rs10) by filtering this table using
<code>CS_in == TRUE</code>.</p>
<pre class="r"><code>Ex1_Forward_fit$mJAM_marg_est %&gt;% head()</code></pre>
<pre><code>## # A tibble: 6 × 4
##   SNP   mJAM_effect mJAM_se mJAM_log10p
##   &lt;chr&gt;       &lt;dbl&gt;   &lt;dbl&gt;       &lt;dbl&gt;
## 1 rs1       0.0544   0.0125     -4.87  
## 2 rs2       0.0206   0.0112     -1.18  
## 3 rs3       0.035    0.0113     -2.71  
## 4 rs4       0.00245  0.0112     -0.0826
## 5 rs5       0.0302   0.0112     -2.15  
## 6 rs6       0.0323   0.0112     -2.41</code></pre>
<p>If you’d like to check out the multi-population marginal estimates
from mJAM, <code>mJAM_marg_est</code> provides the marginal estimates of
all SNPs under the mJAM model. This includes <code>mJAM_effect</code>
(the point estimate), <code>mJAM_se</code> (the standard error of the
point estimate) and <code>mJAM_log10p</code> (log10 of the p-value based
on point estimate and standard error).</p>
</div>
<div id="mjam-susie" class="section level3">
<h3>mJAM-SuSiE</h3>
<pre class="r"><code># --- Fit mJAM_SuSiE
Ex1_SuSiE_fit &lt;- mJAM_SuSiE(marginal.betas = list(MargBeta$MargBeta_P1, MargBeta$MargBeta_P2, MargBeta$MargBeta_P3), 
                      marginal.se = list(MargBeta$MargSEBeta_P1, MargBeta$MargSEBeta_P2, MargBeta$MargSEBeta_P3), 
                      EAFs = list(EAF$EAF_P1, EAF$EAF_P2, EAF$EAF_P3), 
                      N_GWAS = c(5000, 5000, 5000),
                      X_ref = list(RefDosage_P1,RefDosage_P2,RefDosage_P3),
                      SNP_names = MargBeta$SNP,
                      SuSiE_num_comp = 10, 
                      SuSiE_coverage = 0.95)
## --- mJAM_SuSiE results
mJAM_SuSiE_get_cs(susie_fit = Ex1_SuSiE_fit$fit, SNP_names = MargBeta$SNP)</code></pre>
<pre><code>##   index  coverage CS_size index_SNP CS_SNP
## 1    L1 0.9558736       2       rs1    rs1
## 2    L1 0.9558736       2       rs1    rs8</code></pre>
<p><code>mJAM_SuSiE</code> outputs a table with 5 columns:
<code>index</code>, <code>coverage</code>,
<code>CS_size</code>,<code>index_SNP</code> and <code>CS_SNP</code>.</p>
<ul>
<li><code>index</code> is a label for a distinct credible set.</li>
<li><code>coverage</code> is the empirical coverage of this credible
set.</li>
<li><code>CS_size</code> shows how many SNPs in total are there in this
credible set.</li>
<li><code>index_SNP</code> is the name of the index SNP (SNP with
highest posterior probability) in this credible set.</li>
<li><code>CS_SNP</code> is names of individual SNPs selected in this
credible set.</li>
</ul>
<p>In this example, mJAM-SuSiE identifies one credible set with
<code>rs1</code> and <code>rs8</code> in it. Both of them are included
in the mJAM-Forward credible set as well.</p>
</div>
</div>
<div id="example-2-missing-snp-info-in-some-populations"
class="section level2 tabset">
<h2 class="tabset">Example 2: Missing SNP info in some populations</h2>
<p>In the second example, we will explore the scenario when we have
missing data in marginal summary statistics and in the reference dosage.
The dataset used in example 2 is the same as in example 1, except that
there are 2 SNPs made missing in either summary statistics or reference
dosage: the dosage of rs15 in population 1 is missing; the summary
statistic of rs20 in population 3 is missing.</p>
<p>Now let’s load the data:</p>
<pre class="r"><code># Load marginal betas data 
MargBeta &lt;- read.table(&quot;./Example-Data/Marg_2.txt&quot;, 
                       colClasses = c(&quot;character&quot;, rep(&quot;numeric&quot;, 6)))
names(MargBeta) &lt;- c(&quot;SNP&quot;, &quot;MargBeta_P1&quot;, &quot;MargSEBeta_P1&quot;, 
                     &quot;MargBeta_P2&quot;, &quot;MargSEBeta_P2&quot;,
                     &quot;MargBeta_P3&quot;, &quot;MargSEBeta_P3&quot;)

# Take a look at SNP No.20-25, rs20 in P3 missing
MargBeta[20:25,]</code></pre>
<pre><code>##     SNP  MargBeta_P1 MargSEBeta_P1 MargBeta_P2 MargSEBeta_P2  MargBeta_P3
## 20 rs20  0.001697950    0.02514703  0.01435850    0.02056006           NA
## 21 rs21 -0.010286390    0.02488344 -0.06636717    0.02066592 -0.009864962
## 22 rs22  0.035685926    0.02539317  0.03215446    0.02055493 -0.034835993
## 23 rs23  0.031534739    0.02484755  0.04407369    0.02058289 -0.026641422
## 24 rs24 -0.002939056    0.02526160  0.01915812    0.02059609 -0.008948019
## 25 rs25 -0.005859474    0.02549730 -0.01123426    0.02081805 -0.027068635
##    MargSEBeta_P3
## 20            NA
## 21    0.02024422
## 22    0.02054327
## 23    0.02026291
## 24    0.02053724
## 25    0.02019436</code></pre>
<pre class="r"><code># Load effective allele frequency data
EAF &lt;- read.table(&quot;./Example-Data/EAF_2.txt&quot;,
                  colClasses = c(&quot;character&quot;, rep(&quot;numeric&quot;, 3)))
names(EAF) &lt;- c(&quot;SNP&quot;, &quot;EAF_P1&quot;, &quot;EAF_P2&quot;, &quot;EAF_P3&quot;)

# Take a look at SNP No.20-25, rs20 in P3 missing
EAF[20:25, ]</code></pre>
<pre><code>##     SNP EAF_P1 EAF_P2 EAF_P3
## 20 rs20 0.1957 0.3927     NA
## 21 rs21 0.2046 0.3954 0.5990
## 22 rs22 0.1999 0.3968 0.5998
## 23 rs23 0.1995 0.4047 0.5998
## 24 rs24 0.1974 0.3973 0.6099
## 25 rs25 0.1979 0.4015 0.5992</code></pre>
<p>Note that rs20’s <code>beta</code>,<code>se</code>, and
<code>EAF</code> are missing in population 3.</p>
<pre class="r"><code># Load individual-level SNP data from reference panel
RefDosage_P1 &lt;- read.table(&quot;./Example-Data/Dosage_S1_2.txt&quot;, header = TRUE)
RefDosage_P2 &lt;- read.table(&quot;./Example-Data/Dosage_S2_2.txt&quot;, header = TRUE)
RefDosage_P3 &lt;- read.table(&quot;./Example-Data/Dosage_S3_2.txt&quot;, header = TRUE)

# Notice that rs15 in P1 is missing
length(names(RefDosage_P1)); &quot;rs15&quot; %in% names(RefDosage_P1)</code></pre>
<pre><code>## [1] 49</code></pre>
<pre><code>## [1] FALSE</code></pre>
<pre class="r"><code>length(names(RefDosage_P2)); &quot;rs15&quot; %in% names(RefDosage_P2)</code></pre>
<pre><code>## [1] 50</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="r"><code>length(names(RefDosage_P3)); &quot;rs15&quot; %in% names(RefDosage_P3)</code></pre>
<pre><code>## [1] 50</code></pre>
<pre><code>## [1] TRUE</code></pre>
<p>Note that rs15 is missing in the reference panel of population 1.</p>
<p><strong>Note on input formatting requirement</strong>:</p>
<p>When combining summary data from multiple populations, remember to
put <code>NA</code> into where the SNP information is missing in some
populations but present in at least one other population. For SNPs that
are missing in ALL populations, just exclude them.</p>
<pre class="r"><code># Fill in rs15 as NAs
RefDosage_P1 &lt;- cbind(RefDosage_P1[,1:14], NA, RefDosage_P1[,15:49])
names(RefDosage_P1) &lt;- names(RefDosage_P2)</code></pre>
<p>Now we’ll use the marginal summary statistics (<code>beta</code>,
<code>se</code>, <code>EAF</code> and <code>reference dosage</code>) to
run <code>mJAM-SuSiE</code> and <code>mJAM-SuSiE</code>.</p>
<div id="mjam-forward-1" class="section level3">
<h3>mJAM-Forward</h3>
<pre class="r"><code>Ex2_Forward_fit &lt;- mJAM_Forward(N_GWAS = c(5000, 5000, 5000),
                                X_ref = list(RefDosage_P1,RefDosage_P2,RefDosage_P3),
                                Marg_Result = MargBeta,
                                EAF_Result = EAF,
                                condp_cut = 0.05/50, 
                                within_pop_threshold = 0.50,
                                across_pop_threshold = 0.20)</code></pre>
<pre><code>## No. 1 selected index SNP: rs1</code></pre>
<pre><code>## 10 SNPs got pruned. 41 SNPs left.</code></pre>
<pre><code>## Continue to next round of index SNP selection.</code></pre>
<pre><code>## Analysis ended.</code></pre>
<pre class="r"><code>## --- mJAM Forward results: index SNPs  
Ex2_Forward_fit$index</code></pre>
<pre><code>## # A tibble: 1 × 6
##   SNP   b_joint b_joint_var cond_log10p final_log10p  pcut
##   &lt;chr&gt;   &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
## 1 rs1    0.0544    0.000156       -4.86        -4.86 0.001</code></pre>
<pre class="r"><code>## --- mJAM Forward results: credible set SNPs
Ex2_Forward_fit$cs %&gt;% filter(CS_in == T)</code></pre>
<pre><code>## # A tibble: 3 × 7
##   index_SNP CS_SNP Post_Model_Prob Post_Med_Prob SD_Post_CS_Prob CumSum_…¹ CS_in
##   &lt;chr&gt;     &lt;chr&gt;            &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;     &lt;dbl&gt; &lt;lgl&gt;
## 1 rs1       rs1             1              1              0.768      0.768 TRUE 
## 2 rs1       rs8             0.207          0.826          0.131      0.899 TRUE 
## 3 rs1       rs10            0.0859         0.850          0.0561     0.955 TRUE 
## # … with abbreviated variable name ¹​CumSum_Prob</code></pre>
<p>In this toy example, missing values do not affect
<code>mJAM_Forward</code>’s results.</p>
</div>
<div id="mjam-susie-1" class="section level3">
<h3>mJAM-SuSiE</h3>
<pre class="r"><code># --- Fit mJAM_SuSiE, with missing SNPs. 
Ex2_SuSiE_fit &lt;- mJAM_SuSiE(marginal.betas = list(MargBeta$MargBeta_P1, MargBeta$MargBeta_P2, MargBeta$MargBeta_P3), 
                       marginal.se = list(MargBeta$MargSEBeta_P1, MargBeta$MargSEBeta_P2, MargBeta$MargSEBeta_P3), 
                       EAFs = list(EAF$EAF_P1, EAF$EAF_P2, EAF$EAF_P3), 
                       N_GWAS = c(5000, 5000, 5000),
                       X_ref = list(RefDosage_P1,RefDosage_P2,RefDosage_P3),
                       p_cases = c(0.5, 0.5, 0.5), 
                       SNP_names = MargBeta$SNP,
                       SuSiE_num_comp = 10, 
                       SuSiE_coverage = 0.95)

## --- mJAM_SuSiE results
mJAM_SuSiE_get_cs(susie_fit = Ex2_SuSiE_fit$fit, SNP_names = MargBeta$SNP) </code></pre>
<pre><code>##   index  coverage CS_size index_SNP CS_SNP
## 1    L1 0.9560648       2       rs1    rs1
## 2    L1 0.9560648       2       rs1    rs8</code></pre>
<p>In this toy example, missing values do not affect
<code>mJAM_SuSiE</code>’s results.</p>
</div>
</div>
<div id="example-3-mjam-forward-with-pre-specified-index"
class="section level2">
<h2>Example 3: mJAM-Forward with pre-specified index</h2>
<p>One practical advantage of mJAM-Forward is that it provides the
flexibility to pre-specify your own index SNP by using the option
<code>index_snps</code> in <code>mJAM_Forward()</code>. This will bypass
the step of index SNP selection but still construct credible sets of the
specified index SNPs. In the next example, we will specify two imaginary
index SNPs: <code>rs2</code> and <code>rs11</code> by adding
<code>index_snps = c("rs2", "rs11")</code> to
<code>mJAM_Forward()</code>.</p>
<p>Note that the order of <code>index_snps</code> matters: if we put
<code>index_snps = c("rs2", "rs11")</code>, <code>mJAM_Forward</code>
will first construct the credible set for <code>rs2</code> and then
prune out SNPs highly correlated with <code>rs2</code>. Then it will
construct the credible set for <code>rs11</code> and form credible set
of <code>rs11</code> by conditioning on the effect of <code>rs2</code>.
If you’d like to get unconditional credible sets for <code>rs2</code>
and <code>rs11</code> separately, we recommend running
<code>mJAM_Forward</code> twice where specifying one
<code>index_snp</code> at a time.</p>
<pre class="r"><code>Ex3_Forward_fit &lt;- mJAM_Forward(N_GWAS = c(5000, 5000, 5000),
                                X_ref = list(RefDosage_P1,RefDosage_P2,RefDosage_P3),
                                Marg_Result = MargBeta,
                                EAF_Result = EAF,
                                condp_cut = 0.05/50, 
                                index_snps = c(&quot;rs2&quot;, &quot;rs11&quot;), 
                                within_pop_threshold = 0.50,
                                across_pop_threshold = 0.20)</code></pre>
<pre><code>## No. 1 selected index SNP: rs2</code></pre>
<pre><code>## 2 SNPs got pruned. 49 SNPs left.</code></pre>
<pre><code>## Continue to next round of index SNP selection.</code></pre>
<pre><code>## No. 2 selected index SNP: rs11</code></pre>
<pre><code>## 3 SNPs got pruned. 49 SNPs left.</code></pre>
<pre><code>## Continue to next round of index SNP selection.</code></pre>
<pre><code>## Analysis ended.</code></pre>
<pre class="r"><code>## --- mJAM Forward results: index SNPs  
Ex3_Forward_fit$index</code></pre>
<pre><code>## # A tibble: 2 × 6
##   SNP   b_joint[,1] b_joint_var cond_log10p final_log10p  pcut
##   &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
## 1 rs2        0.0210    0.000125       -1.19        -1.22 0.001
## 2 rs11      -0.0286    0.000125       -1.63        -1.63 0.001</code></pre>
<pre class="r"><code>## --- mJAM Forward results: credible set SNPs
Ex3_Forward_fit$cs %&gt;% filter(CS_in == T) </code></pre>
<pre><code>## # A tibble: 15 × 7
##    index_SNP CS_SNP Post_Model_Prob Post_Med_Prob SD_Post_CS_Prob CumSum…¹ CS_in
##    &lt;chr&gt;     &lt;chr&gt;            &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;    &lt;dbl&gt; &lt;lgl&gt;
##  1 rs2       rs2              1           1               0.394      0.394 TRUE 
##  2 rs2       rs1              1           0.525           0.207      0.601 TRUE 
##  3 rs2       rs5              1           0.177           0.0697     0.671 TRUE 
##  4 rs2       rs6              1           0.159           0.0625     0.733 TRUE 
##  5 rs2       rs7              0.534       0.297           0.0624     0.795 TRUE 
##  6 rs2       rs3              1           0.137           0.0540     0.849 TRUE 
##  7 rs2       rs9              1           0.0854          0.0337     0.883 TRUE 
##  8 rs2       rs4              0.186       0.417           0.0305     0.914 TRUE 
##  9 rs2       rs10             1           0.0715          0.0282     0.942 TRUE 
## 10 rs2       rs8              1           0.0514          0.0203     0.962 TRUE 
## 11 rs11      rs11             1           1               0.913      0.913 TRUE 
## 12 rs11      rs26             1           0.0147          0.0134     0.927 TRUE 
## 13 rs11      rs9              1           0.0101          0.00925    0.936 TRUE 
## 14 rs11      rs6              0.708       0.0132          0.00853    0.944 TRUE 
## 15 rs11      rs35             1           0.00822         0.00751    0.952 TRUE 
## # … with abbreviated variable name ¹​CumSum_Prob</code></pre>
<p>In the above example, mJAM credible set for <code>rs2</code> consists
of all 10 SNPs in the same LD block as <code>rs2</code>. Conditional on
<code>rs2</code>, the credible set for <code>rs11</code> consists of 5
SNPs (all within the same LD block as <code>rs11</code>).</p>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 4.2.0 (2022-04-22)
## Platform: aarch64-apple-darwin20 (64-bit)
## Running under: macOS 13.0.1
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] dplyr_1.1.0       hJAM_1.0.1        fontawesome_0.3.0
## 
## loaded via a namespace (and not attached):
##  [1] shape_1.4.6      tidyselect_1.2.0 xfun_0.32        bslib_0.4.0     
##  [5] reshape2_1.4.4   splines_4.2.0    lattice_0.20-45  colorspace_2.1-0
##  [9] vctrs_0.5.2      generics_0.1.3   htmltools_0.5.3  yaml_2.3.5      
## [13] gmp_0.7-1        utf8_1.2.3       survival_3.3-1   rlang_1.0.6     
## [17] jquerylib_0.1.4  pillar_1.8.1     glue_1.6.2       Rmpfr_0.9-1     
## [21] withr_2.5.0      foreach_1.5.2    lifecycle_1.0.3  plyr_1.8.8      
## [25] stringr_1.5.0    munsell_0.5.0    gtable_0.3.1     codetools_0.2-18
## [29] evaluate_0.16    knitr_1.40       fastmap_1.1.0    fansi_1.0.4     
## [33] Rcpp_1.0.10      scales_1.2.1     susieR_0.10.0    cachem_1.0.6    
## [37] jsonlite_1.8.0   ggplot2_3.4.1    digest_0.6.29    stringi_1.7.12  
## [41] grid_4.2.0       cli_3.6.0        tools_4.2.0      magrittr_2.0.3  
## [45] sass_0.4.2       glmnet_4.1-6     tibble_3.1.8     pkgconfig_2.0.3 
## [49] Matrix_1.4-1     matrixcalc_1.0-6 rmarkdown_2.14   reshape_0.8.9   
## [53] rstudioapi_0.14  iterators_1.0.14 R6_2.5.1         compiler_4.2.0</code></pre>
</div>

<br><br>
<center>
<footer 
    display: flex;
    justify-content: space-between;
    text-align: justify >
  <p class="footerP"> © 2022 University of Southern California. All rights reserved.
This site is hosted by Github Pages and is built on R Markdown. </p>
</footer>

</center>





</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
