---
title: "Fine-Mapping and Credible Set Construction using a Multi-population Joint Analysis of Marginal Summary Statistics from Genome-wide Association Studies"
abstract: | 
  Recent advancement in Genome-wide Association Studies (GWAS) comes from not only increasingly larger sample sizes but also the shifted focus towards underrepresented populations. Multi-population GWAS may increase power to detect novel risk variants and improve fine-mapping resolution by leveraging evidence from diverse populations and accounting for the difference in linkage disequilibrium (LD) across ethnic groups. Here, we expand upon our previous approach for single-population fine-mapping through Joint Analysis of Marginal SNP Effects (JAM) to a multi-population analysis (mJAM). Under the assumption that true causal variants are common across studies, we implement a novel version of JAM that conditions on multiple SNPs while explicitly incorporating the different LD structures across populations. The mJAM framework can be used to first select index variants using the mJAM likelihood with any feature selection approach. In addition, we present a novel approach leveraging the ideas of mediation to construct credible sets for these index variants. Construction of such credible sets can be performed given any existing index variants. We illustrate the implementation of the mJAM likelihood through two implementations: mJAM-SuSiE (a Bayesian approach) and mJAM-Forward selection. Through simulation studies based on realistic effect sizes and levels of LD, we demonstrated that mJAM performs better than other existing multi-ethnic methods for constructing concise credible sets that include the underlying causal variants. In real data examples taken from the most recent multi-population prostate cancer GWAS, we showed several practical advantages of mJAM over other existing methods.
---

### Manuscript
Preprint is now available on bioariv: https://www.biorxiv.org/content/10.1101/2022.12.22.521659v1

### Software availability

[mJAM is available on Github as part of the 'hJAM' package](https://github.com/USCbiostats/hJAM)

